### PR TITLE
ref(chartutil): introduce chartutil.Save

### PR DESCRIPTION
**Pull request by bacongobbler**
chartutil.Save has been renamed to chartutil.SaveArchive.

chartutil.Save accepts a chart and an io.Writer, wrapping the logic around serializing a chart to an io.Writer.

This allows other packages to write higher-level abstractions above chart.Save. For example, streaming a chart to a *bytes.Buffer.

Signed-off-by: Matthew Fisher <matt.fisher@microsoft.com>

### CODE REVIEW

**Review by mattfarina — COMMENTED** [review]
It may be overkill but should there be another test for just `Save`?

**Comment by adamreese** [comment]
Do the verbs import and export help to clarify what each function is doing?

**Comment by bacongobbler** [comment]
> Do the verbs import and export help to clarify what each function is doing?

Perhaps? `.Save` seems fairly clear to me the intent of the function so it's not the verbs that are tripping me up, but the suffix after the verb. Take `SaveDir` for example: does that archive a chart as a .tgz and write that to a directory, or does it write the contents of a chart in a directory? The naming is slightly ambiguous, causing a little bit of confusion based on the name, so I have to look at the code to inspect what is the function's intended use.

Looking through the code, we have several functions that all deal with charts that are archived. I'll give this package a bit more thought on the design and how we want to address charts coming from multiple different formats (.zip, .bzip, .xz, for example).

**Comment by bacongobbler** [comment]
> It may be overkill but should there be another test for just `Save`?

If history has taught us anything, more tests are always better :)

File: pkg/chartutil/save.go [hunk 1]
```diff
@@ -33,6 +34,11 @@ import (
 var headerBytes = []byte("+aHR0cHM6Ly95b3V0dS5iZS96OVV6MWljandyTQo=")
 
 // SaveDir saves a chart as files in a directory.
+//
+// This takes an existing chart and a destination directory.
+//
+// If the directory is /foo, and the chart is named bar, with version 1.0.0, this
```
Comment (thomastaylor312) [inline pkg/chartutil/save.go]: Nit: The fact that this chart is version 1.0.0 seems to be irrelevant to the rest of the comment and can probably be dropped

**Review by thomastaylor312 — APPROVED** [review]
Super tiny, non-blocking nit. Otherwise this is a great addition

**Comment by bacongobbler** [comment]
closing as irrelevant for now
